### PR TITLE
feat(docs): Improve documentation on filtering values.

### PR DESCRIPTION
**Pull request by aquamatthias**
# Description

# To-Dos

- [x] Format with `yarn format`
- [x] Lint with `yarn lint`
- [x] Build successfully with `yarn build`

# Issues Fixed

- Fixes #XXXX

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

**Review by meln1k — DISMISSED** [review]

File: docs/concepts/search/filters.md [hunk 1]
```diff
@@ -23,13 +23,37 @@ In order to filter for specific attributes of a node, it is possible to define p
...
+{ "some": { "nested": [1, 2, 3] } }
+```
+
+We can use the path `some.nested[0]` to access the first element of the array, and `some.nested[2]` to access the third element.
+
+If the array contains objects, we can access the specific element by using the index notation: `some.nested[0].prop`.
+
+```json
+{ "some": { "nested": [{ "prop": 1 }, { "prop": 2 }, { "prop": 3 }] } }
+```
```
Comment (TheCatLady) [inline docs/concepts/search/filters.md:50]:
````suggestion
#### Arrays

If a property is an array, specific elements can be accessed using index notation.

In the below JSON, the path `some.nested[0]` references first element of the array with path `some.nested`, and `some.nested[2]` points to the third (and last) element:

```json
{ "some": { "nested": [1, 2, 3] } }
```

In arrays of objects, object elements can also be accessed using index notation.

`some.nested[0].prop` has a value of `1` here:

```json
{ "some": { "nested": [{ "prop": 1 }, { "prop": 2 }, { "prop": 3 }] } }
```
````

File: docs/concepts/search/filters.md [hunk 2]
```diff
@@ -23,13 +23,37 @@ In order to filter for specific attributes of a node, it is possible to define p
...
+
+In order to select elements, where the first nested prop is bigger than 2, we can use the path `some.nested[0].prop > 2`. If we want to select elements that have any nested prop bigger than 2, we can use the path `some.nested[*].prop > 2`. The `*` wildcard is used to perform the filter criteria on all elements of the array, where only one match would be sufficient to match.
```
Comment (TheCatLady) [inline docs/concepts/search/filters.md:52]: I don't quite understand this portion. Could you clarify the meaning or give an example?

File: docs/concepts/search/filters.md [hunk 3]
```diff
@@ -23,13 +23,37 @@ In order to filter for specific attributes of a node, it is possible to define p
...
 
-If all relative paths are interpreted relative to `reported`, we need a way to target properties not in the reported section. This is possible by using the absolute path syntax via the `/` (slash).
+### Property sections
 
-A property path that starts with a slash is always interpreted absolute.
+Most of the properties are defined in the reported section. The CLI interprets all defined property paths relative to the reported section by default. Thus, the path to property `reported.name` can simply be written as `name`. When every path is interpreted relative to `reported`, we need a way to target properties not in the reported section. This is possible by using the absolute path syntax via the `/` (slash). A property path that starts with a slash is always interpreted absolute.
 
 In order to access properties outside of the reported section, use the `/` syntax:
```
Comment (TheCatLady) [inline docs/concepts/search/filters.md:58]:
```suggestion
Most properties are defined in the `reported` section, and the CLI interprets property paths relative to the `reported` section by default. Thus, the path to property `reported.name` can simply be written as `name`.

To target properties not in the `reported` section, prefix the property path with a slash (`/`) for it to be interpreted as an absolute path:
```

**Review by TheCatLady — CHANGES_REQUESTED** [review]

**Review by TheCatLady — APPROVED** [review]

**Comment by TheCatLady** [comment]
@aquamatthias Please see if I described the idea correctly. If so, feel free to merge!

**Pull request merged**